broker-daemon/interchain-router/get-preimage.js

```diff
@@ -108,16 +107,21 @@ async function getPreimage ({ params, send, onCancel, onError, ordersByHash, eng
     return send({ permanentError: err })
   }
 
-  let timeLockDelta
   try {
-    timeLockDelta = timeLockDeltaInSeconds(inboundEngine, timeLock, bestHeight)
+    var timeLockDelta = timeLockDeltaInSeconds(inboundEngine, timeLock, bestHeight)
   } catch (err) {
     logger.error(err)
     return send({ permanentError: err.message })
   }
 
   logger.debug(`Sending payment to ${takerAddress} to translate swap ${swapHash}`)
-  const { paymentPreimage, permanentError } = await outboundEngine.translateSwap(takerAddress, swapHash, outboundAmount, timeLockDelta)
+
+  try {
+    var { paymentPreimage, permanentError } = await outboundEngine.translateSwap(takerAddress, swapHash, outboundAmount, timeLockDelta)
+  } catch (err) {
+    logger.error(err)
+    return send({ permanentError: err.message })
```

Comment:
you can only handle this as a permanent error if you can be certain that the error is permanent. Is that true in this case?

Reply:
We cant guarantee it. Im going to make a change in lnd-engine for the situation I found. The error we want to handle is when no routes are found... however it looks like the call throws an error instead of returning an empty array for `routes`